### ISSUE TITLE
Data preparation

### ISSUE THREAD

**Issue by natasasdj** [raghavchalapathy/oc-nn_old#6]
From your code it is not at all clear how you prepare the cifar-10 data for the modeling. For example, 
in your comment of the code shown bellow, there is a comment "extract the class label from the image path and update the labels list". However, there is no explanation how you save the cifer-10 data into "image paths" in order to differentiate the labels. Moreover, for the training of the OC-NN model, only images with dogs should be used. However, from your code I don't see anywhere where you split the data into dogs and cats, and then save it into some "image path" to use it later.
Could you please clarify the exact steps how the cifar-10 data are prepared and saved to different "image paths" according to different labels? Going through load_dataset.py, none of the given functions doesn't do this.

For making auto-encoder, are all images in cifar-10 used, or only images with both dogs and cats, or only images with dogs? Btw, there is no code for making the auto-encoder model.

I find the code here very incomplete and I am very frustrated unsuccessfully trying to reproduce the results in the paper. Has anyone here succeeded to reproduced the claimed results? 

```
imagePaths = sorted(list(paths.list_images(train_path)))
    # loop over the input training images
    image_dict = {}
    test_image_dict = {}
    i = 0
    for imagePath in imagePaths:
        # load the image, pre-process it, and store it in the data list
        image = cv2.imread(imagePath)
        image = cv2.resize(image, (side, side))
        image = img_to_array(image)
        data.append(image)
        image_dict.update({i: imagePath})
        i = i + 1

        # extract the class label from the image path and update the
        # labels list
        label = imagePath.split(os.path.sep)[-2]
        label = 1 if label == "dogs" else 0
        labels.append(label)
    # scale the raw pixel intensities to the range [0, 1]
    data = np.array(data)

y = keras.utils.to_categorical(labels, num_classes=2)
```
